### PR TITLE
Render questions using filesystem

### PR DESCRIPTION
**Pull request by Crola1702**
# Description

Instead of creating a Nanoc DataSource to load all questions from a json dump, we want to use Filesystem data source and render all questions, answers and comments from the question metadata.

# Changes

This PR:
* Removes outdated compilation rules to keep only `/question/*.md` rule.
* Removes `questions` DataSource
* Adds a sample markdown question: [:scroll: 1.md](https://github.com/ros-infrastructure/answers-archive/blob/markdown-questions-dump/content/question/1.md?plain=1).
* Adds a helper function to render markdown from a string.
  > **Note**: This was the way I managed to render answers and comments `BodyMarkdown`, as they're not generated as items, so we can't apply redcarpet filter.
* Update layouts to render answers and comments based in question item metadata

I see answers.gazebosim stores all the questions under `/question/:id` endpoint, I believe it is better to have the same directory structure, so I created `content/question/` also.

Example output of `1.md` [gazebosim-1-html.pdf](https://github.com/user-attachments/files/16380370/gazebosim-1-html.pdf)
> GitHub doesn't allow adding html files

### CODE REVIEW

**Comment by Crola1702** [comment]
@nuclearsandwich I recall talking about compiling two times, one for gazebosim and other for ros? How is the website split idea?

File: content/question/1.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+MigratedUrl: https://robotics.stackexchange.com/questions/25232/how-to-stop-the-joint-motion-clear-the-force
+OriginalPostID: 1
+OriginalPostURL: https://answers.gazebosim.org/question/1/how-to-stop-the-joint-motion(-clear-the-force)/
+OwnerDisplayName: nkoenig
+OwnerKarma: 7676
+OwnerProfileUrl: https://answers.gazebosim.org/users/5/nkoenig/
+Tags: gazebo
+Title: how to stop the joint motion( clear the force)
+UpvoteCount: 0
+---
```
Comment (nuclearsandwich) [inline content/question/1.md:104]: I'm very tempted to ask for metadata to be in in `snake_case` format matching Ruby / Python identifier conventions. If that's immeasurably hard then we'll roll with this.

File: layouts/answer.html [hunk 2]
```diff
@@ -1,22 +1,18 @@
-<div class="answer<% if item[:accepted?] %> accepted<% end %>">
+<div class="answer<% if item[:Accepted] %> accepted<% end %>">
   <%= yield %>
-  <% if item[:accepted?] %>
+  <% if item[:Accepted] %>
   <div class="accepted">
-    <span class="accepted">☑️</span> This answer was <strong>ACCEPTED</strong> on the original site.
+    <span class="accepted">☑️ </span> This answer was <strong>ACCEPTED</strong> on the original site.
```
Comment (nuclearsandwich) [inline layouts/answer.html]: Why is a space needed inside the span with the emoji rather than only outside it?

File: layouts/comment.html [hunk 3]
```diff
@@ -1,6 +1,4 @@
 <div class="comment">
   <%= yield %>
-  <div class="owner">
-    <p>Comment by <%= item[:owner].display_name %> on <%= pretty_date item[:creation_date] %></p>
-  </div>
+  <div class="owner"><p>Asked by <%= item[:OwnerDisplayName] %> on <%= pretty_date item[:CreationDate] %></p></div>
```
Comment (nuclearsandwich) [inline layouts/comment.html]: TODO: we'll want this to eventually be a link to the author's profile (index of questions asked and answered)

File: content/question/1.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    \ become constant. With frictions it will decellerate, with repeated force the\
+    \ arm will accellerate.\r\nwith repeated force and friction the arm will reach\
+    \ a steady state where you'r applied force equals the force from friction, and\
+    \ from here the speed will eventually become constant.\r\n\r\n**solutionset:**\r\
+    \n\r\n1. if you are repeatingly applying force, stop that after some time and\
+    \ let the friction stop the arm (use gazebos simulation time..)\r\n\r\n2. learn\
+    \ control theory, (and basic physics) and let a controller that calculate the\
+    \ desired force to have the arm move to a position or speed or both be implemented.\
+    \  (You should do this to control the HW anyway)\r\n\r\n3. stop using forces,\
+    \ _cheat_ and use the animation interface for gazebo\r\n"
```
Comment (nuclearsandwich) [inline content/question/1.md]: This format parses but it isn't the literal scalar format we discussed. Was it not possible to generate scalar literals after all?

File: lib/helpers.rb [hunk 5]
```diff
@@ -47,6 +47,12 @@ def migrated_gz_posts
   end
 end
 
+def render_markdown md_str
+  # This is inefficient as markdown renderer should not be created every time
+  markdown = Redcarpet::Markdown.new(Redcarpet::Render::HTML)
+  markdown.render(md_str)
+end
+
```
Comment (nuclearsandwich) [inline lib/helpers.rb]: I'm pushing a commit that I think helps here.

File: content/question/1.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  OwnerDisplayName: adaman05
+  OwnerKarma: 1
+  OwnerProfileUrl: https://answers.gazebosim.org/users/3828/adaman05/
+  UpvoteCount: 0
+Comments:
+- BodyMarkdown: i have the same kind of problem ,i tried to modify the c++ code
+    in tutorial http://gazebosim.org/wiki/Tutorials/1.3/control_robot/mobile_base_laser
+    .it also not stopping.i have setForce to zero even it is not stooping .I am
+    beginner in field of robotics as well as in gazebo .plz guide me ,i am attachin
+    my modified code here
```
Comment (nuclearsandwich) [inline content/question/1.md]: It's weird that this is unqouted and still seems to work fine. I expect because of the indentation.

Comment (nuclearsandwich) [inline content/question/1.md]: Is this comment copied here to test the layout? It's not here on the current Answers site: https://answers.gazebosim.org/question/1/how-to-stop-the-joint-motion-clear-the-force/

**Comment by nuclearsandwich** [comment]
> I recall talking about compiling two times, one for gazebosim and other for ros? How is the website split idea?

My uncomplicated idea is basically to maintain `gazebo` and `ros` branches of this repository, each with the data for the respective site and cherry-pick shared changes between them.

A more complicated idea is to store no content in the `main` branch and add helper scripts to check `ros` or `gz` content out before a build.

File: content/question/1.md (hunk 1 above)
Comment (Crola1702) [reply content/question/1.md:104]: Yeah that's doable. Will take a look

File: layouts/answer.html (hunk 2 above)
Comment (Crola1702) [reply layouts/answer.html]: Good question. I don't know where did it come from :thinking:

File: layouts/comment.html (hunk 3 above)
Comment (Crola1702) [reply layouts/comment.html]: Should we also generate author pages?

File: content/question/1.md (hunk 4 above)
Comment (Crola1702) [reply content/question/1.md]: I'm having a weird problem there. Some questions are generated as scalar literals and some are not. I'm still debugging why is this happening

File: lib/helpers.rb (hunk 5 above)
Comment (Crola1702) [reply lib/helpers.rb]: Do you think there is a better way to render markdown in metadata? I thought about using a module or class and create only one renderer, but for that it seems I need esoteric ruby.

File: content/question/1.md (hunk 6 above)
Comment (Crola1702) [reply content/question/1.md]: That's true. I'll update this post in a new dump example version

Comment (Crola1702) [reply content/question/1.md]: Yeah, good catch! I was testing everything worked properly, so I copied and pasted a comment.

File: content/question/1.md (hunk 4 above)
Comment (Crola1702) [reply content/question/1.md]: Just to save every step I take from here:

If the dumped string contains a "\r" character, the string is dumped in quotes (that's my investigation for now, I'm not sure why is that.

File: content/question/1.md (hunk 1 above)
Comment (Crola1702) [reply content/question/1.md:104]: Done in 9bfef93

File: layouts/answer.html (hunk 2 above)
Comment (Crola1702) [reply layouts/answer.html]: Removed in 7e1fd06

File: content/question/1.md (hunk 4 above)
Comment (Crola1702) [reply content/question/1.md]: Now it seems everything has the literal now. (Just replaced "\r" for "" in all body markdowns)

File: content/question/1.md (hunk 6 above)
Comment (Crola1702) [reply content/question/1.md]: The new question example ins updated now

Comment (Crola1702) [reply content/question/1.md]: Updated in 9bfef93

**Review by nuclearsandwich — APPROVED** [review]

**Pull request merged**